File: src/main/java/dev/bstk/utils/StringUtils.java
package dev.bstk.utils;

public class StringUtils {

    private StringUtils() { }

    public static final String S = "S";
    public static final String N = "N";

    public static boolean isEmpty(final String valor) {
        return valor == null
            || valor.isEmpty()
            || valor.isBlank();
    }

    public static boolean isNotEmpty(final String valor) {
        return !isEmpty(valor);
    }

    public static boolean isSN(final String valor) {
        return isNotEmpty(valor)
            && (S.equalsIgnoreCase(valor) || N.equalsIgnoreCase(valor));
    }

    public static boolean isNotSN(final String valor) {
        return !isSN(valor);
    }

}

File: src/main/kotlin/dev/bstk/StringUtils.kt
package dev.bstk

const val S = "S"
const val N = "N"

fun isSN(valor: String): Boolean {
  return valor.isNotEmpty()
      && valor.isNotBlank()
      && (valor.equals(S, true) || valor.equals(N, true))
}

fun isNotSN(valor: String): Boolean {
  return !isSN(valor)
}
